src/Microsoft.ML.DataView/TypeManager.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.ML.Internal.DataView;
+
+namespace Microsoft.ML.Data
+{
+    public static class TypeManager
+    {
+        // Types have been used in ML.NET type systems. They can have multiple-to-one type mapping.
+        // For example, UInt32 and Key can be mapped to uint. We enforce one-to-one mapping for all
+        // user-registered types.
+        private static HashSet<Type> _notAllowedRawTypes;
+        private static ConcurrentDictionary<Type, DataViewType> _rawTypeToDataViewTypeMap;
+        private static ConcurrentDictionary<DataViewType, Type> _dataViewTypeToRawTypeMap;
```

Comment:
these can all be made 'static readonly' I think.

Reply:
This will be removed and the initializers of static fields will be used.

---
In reply to: [285262825](https://github.com/dotnet/machinelearning/pull/3263#discussion_r285262825) [](ancestors = 285262825)